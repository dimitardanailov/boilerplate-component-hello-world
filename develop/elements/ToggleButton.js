import React, { Component } from 'react';
import { Subject } from 'rxjs';

class ToggleButton extends Component {
	constructor(props) {
		super(props);

		this.state = { isToggleOn: true };

		this.setToggle = new Subject();
	}

	componentDidMount() {
		this.setToggle.subscribe(prevState => {
			this.setState(() => ({
				isToggleOn: !prevState
			}));
		});
	}

	getToogle() {
		return this.state.isToggleOn;
	}

	render() {
		return (
			<div>
				<button onClick={() => this.setToggle.next(this.getToogle())}>
					{this.state.isToggleOn ? 'ON' : 'OFF'}
				</button>
			</div>
		);
	}
}

export default ToggleButton;
