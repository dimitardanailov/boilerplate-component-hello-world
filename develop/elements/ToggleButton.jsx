import React, { Component } from 'react';
import { Subject } from 'rxjs';

class ToggleButton extends Component {
	constructor(props) {
		super(props);

		this.state = { isToggleOn: true };

		this.updateToggleProperty = new Subject();
	}

	componentDidMount() {
		this.updateToggleProperty.subscribe(prevState => {
			this.setState(() => ({
				isToggleOn: !prevState.isToggleOn
			}));
		});
	}

	render() {
		return (
			<div>
				<button onClick={() => this.updateToggleProperty.next(this.state.isToggleOn)}>
					{this.state.isToggleOn ? 'ON' : 'OFF'}
				</button>
			</div>
		);
	}
}

export default ToggleButton;
