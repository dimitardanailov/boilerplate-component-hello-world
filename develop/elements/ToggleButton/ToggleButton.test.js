import React from 'react';
import renderer from 'react-test-renderer';
import ToggleButton from './ToggleButton';

test('ToggleButton - toMatchSnapshot', () => {
	const component = renderer.create(<ToggleButton />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('ToggleButton has a children', () => {
	const component = renderer.create(<ToggleButton />);

	const tree = component.toJSON();
	expect(tree.children).toHaveLength(1);
});

test('ToggleButton button state is equal to ON', () => {
	const component = renderer.create(<ToggleButton />);

	const tree = component.toJSON();
	const button = tree.children[0];

	const defaultState = 'ON';
	expect(button.children[0]).toEqual(defaultState);
});

test('ToggleButton button has onClick() function', () => {
	const component = renderer.create(<ToggleButton />);

	const tree = component.toJSON();
	const button = tree.children[0];

	expect(typeof button.props.onClick).toEqual('function');
});
