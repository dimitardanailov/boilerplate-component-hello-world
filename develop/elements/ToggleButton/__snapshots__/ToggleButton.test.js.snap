// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton - toMatchSnapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    ON
  </button>
</div>
`;
